### ISSUE TITLE
Accessibility Testing for Privacy Act Request VA Form 20-10206

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#70633]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jeana-adhoc 

### Use of color and color contrast checks

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [ ] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

On the disability exam details page, when we put the memorable date component in a 'list and loop' card with a grey background, the hint text fails color contrast rules. 

`Element has insufficient color contrast of 4.06 (foreground color: #71767a, background color: #f1f1f1, font size: 12.7pt (16.96px), font weight: normal). Expected contrast ratio of 4.5:1`

<img width="450" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/2f98e502-d4db-4906-a205-a6e3abc0cdc0">

I think this will be a design system issue.  Related tickets I've found:
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2104
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1693

The other solution, could be to change the grey of the tile card, but it is currently using the [lightest gray](https://design.va.gov/foundation/utilities/background-color) possible, `#f1f1f1` 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/71633

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

| Axe Results  | Page |
| ------------- | ------------- |
| No issues  | [Introduction](https://staging.va.gov/records/request-personal-records-form-20-10206/introduction)  |
| No issues  | [Preparer type](https://staging.va.gov/records/request-personal-records-form-20-10206/preparer-type) |
| No issues | [personal information](https://staging.va.gov/records/request-personal-records-form-20-10206/personal-information) |
| No issues | [non-citizen information](https://staging.va.gov/records/request-personal-records-form-20-10206/non-citizen-identification-information) |
| No issues | [Mailing address](https://staging.va.gov/records/request-personal-records-form-20-10206/contact-information) |
| No issues | [Phone & Email](https://staging.va.gov/records/request-personal-records-form-20-10206/phone-email) |
| No issues | [Records](https://staging.va.gov/records/request-personal-records-form-20-10206/record-selections) |
| Color contrast issues with date hint text on grey card | [Disability exam details](https://staging.va.gov/records/request-personal-records-form-20-10206/disability-exam-details) |
| No issues | [Financial records](https://staging.va.gov/records/request-personal-records-form-20-10206/financial-record-details) |
| No issues | [ Life insurance ](https://staging.va.gov/records/request-personal-records-form-20-10206/life-insurance-benefit-details) |
| No issues | [Compensation/pension details](https://staging.va.gov/records/request-personal-records-form-20-10206/other-compensation-and-pension-details) |
| No issues | [Benefit Records](https://staging.va.gov/records/request-personal-records-form-20-10206/other-benefit-details) |
| Headings out of order | [Review page](https://staging.va.gov/records/request-personal-records-form-20-10206/review-and-submit)

### How did axe testing go?

All pages now have one new error that is out of the forms team control. But this ticket, https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15568 covers the issue.

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] All page elements are readable and usable at 200% zoom
- [X] All page elements are readable and usable at 300% zoom
- [ ] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

- We have an issue with the modal. On the confirmation button "Yes, remove this exam date" the text is not wrapping in the button and it is cut off when the page is zoomed at 400%. 
  - https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/856    
  - <img width="450" alt="Screenshot 2023-12-10 at 3 28 45 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/d8f3384c-106d-423b-83f2-da59ae6c585f">
<br /><br />

- When zooming at 400%, icons overlaps the select text in the select input field
  - **Jeana to talk with a11y COP** 
  - <img width="450" alt="Screenshot 2023-12-10 at 3 28 08 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/4b876067-1db1-483e-b9f0-5c3a4a54329c">
  - This should end up taking care of itself after the [default is converted back to 3 text inputs in this ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2316) 

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [X] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

no issues

### Do you have any other results to share?

- Hint text in the checkbox group v3 is not inside the legend so it is not being read aloud by the screen readers. ([example page](https://staging.va.gov/records/request-personal-records-form-20-10206/record-selections)) 
  - https://github.com/department-of-veterans-affairs/va.gov-team/issues/71634
- Edit & remove buttons on the exam date details page need to be unique. Change `aria-label="Edit exam date"` to `aria-label="Edit exam date [insert the date here]"
   - https://github.com/department-of-veterans-affairs/va.gov-team/issues/71635
- Should there be a character limit displayed for the 'other' field on the [compensation and/or pension details page](https://staging.va.gov/records/request-personal-records-form-20-10206/other-compensation-and-pension-details)?
   - **Question for Rachel** 
- There are several issues with the review page. I think an update has been made that wasn't regression tested.
  - Headings are out of order. The accordions are now h2s, they were previously h3s. 
    - PR By Robin for Platform code: https://github.com/department-of-veterans-affairs/vets-website/pull/27078
  - There's some odd line-height in the review page now.  I think this is a design system issue
  -  **Jeana to talk with dst**
  - <img width="550" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/606a1bd6-23b7-4464-b5e3-cd63a6d0c737">
 

### Items that are issues, but need more research and are not a priority
- The fieldset and legend on the [disability exam details page](https://staging.va.gov/records/request-personal-records-form-20-10206/disability-exam-details) is not being read out by a screen reader. The problem is that the date is already a field in a fieldset and it has its own fieldset and legend... so this is an example of why nested fieldsets don't work so well.  Need to think through if there's enough detail on the date to help a user through.
- On the modal, "Are you sure you want to remove this exam date modal" is announced multiple times, and then the entire modal is read without any user prompt/input. I'm not sure this is the correct behavior. I've seen this come up before, but I don't think we've been able to fix it or understand what is happening.
- On the [personal information page](https://staging.va.gov/records/request-personal-records-form-20-10206/personal-information) "Your date and place of birth" is not announced. But I think this is okay as the date of birth and place of birth are pretty descriptive on their own.

**Comment by jeana-adhoc** [comment]
This is still in progress.

**Comment by jeana-adhoc** [comment]
Summary of tickets:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/71633
- https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/856
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/71634
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/71635

**Comment by jeana-adhoc** [comment]
All of the tickets have been written/created.
